db/db_impl/db_impl_open.cc

```diff
@@ -1929,4 +1929,305 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
   }
   return s;
 }
+
+Status DB::CreateFromDisjointInstances(
+    const MergeInstanceOptions& merge_options, const DBOptions& db_options,
+    const std::string& name,
+    const std::vector<ColumnFamilyDescriptor>& column_families,
+    const std::vector<DB*> instances, std::vector<ColumnFamilyHandle*>* handles,
+    DB** dbptr) {
+  return DBImpl::CreateFromDisjointInstances(merge_options, db_options, name,
+                                             column_families, instances,
+                                             handles, dbptr);
+}
+
+Status DBImpl::CreateFromDisjointInstances(
+    const MergeInstanceOptions& merge_options, const DBOptions& db_options,
+    const std::string& name,
+    const std::vector<ColumnFamilyDescriptor>& column_families,
+    const std::vector<DB*> instances, std::vector<ColumnFamilyHandle*>* handles,
+    DB** dbptr) {
+  Status s;
+  MergeInstanceOptions mopts(merge_options);
+  // Sanity check.
+  if (column_families.size() == 0 ||
+      column_families.front().name !=
+          ROCKSDB_NAMESPACE::kDefaultColumnFamilyName) {
+    return Status::InvalidArgument("Missing default column family descriptor.");
+  }
+  if (!db_options.create_if_missing) {
+    return Status::InvalidArgument(
+        "create_if_missing must be set to create a DB.");
+  }
+  // Create target DB and column families.
+  s = DB::Open(Options(db_options, column_families.front().options), name,
+               dbptr);
+  if (!s.ok()) {
+    delete *dbptr;
```

Comment:
This seems unnecessary as the docs says:

https://github.com/tikv/rocksdb/blob/0b6d58bf0ef7529503cab8dcf9b9ada0b4cf2346/include/rocksdb/db.h#L132-L137